### PR TITLE
Make sequence dictionary validation in BQSR dataflow work when the reference is in a bucket

### PR DESCRIPTION
**Pull request by droazen**

### CODE REVIEW

**Comment by droazen** [comment]
For @jean-philippe-martin to review, please

**Comment by droazen** [comment]
Resolves https://github.com/broadinstitute/hellbender/issues/683

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java [hunk 1]
```diff
@@ -118,7 +121,7 @@ protected void setupPipeline(Pipeline pipeline) {
             BaseRecalibratorDataflowUtils.ensureReferenceIsReadable(pipeline.getOptions(), referencePath);
```
Comment (jean-philippe-martin) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java]: Earlier, on line 116, we say "saveTextualTables = (null!=BRAC.RAC.RECAL_TABLE_FILE)". We'll need to make RECAL_TABLE_FILE optional.

File: src/test/java/org/broadinstitute/hellbender/utils/reference/ReferenceUtilsUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            isClosed = true;
+        }
+
+        public boolean isClosed() {
+            return isClosed;
+        }
+    }
+
+    @Test
+    public void testLoadFastaDictionaryFromStream() throws IOException {
```
Comment (jean-philippe-martin) [inline src/test/java/org/broadinstitute/hellbender/utils/reference/ReferenceUtilsUnitTest.java]: We also need a test that loads from a bucket.

**Comment by jean-philippe-martin** [comment]
Done. It's good to go as soon as these concerns are addressed.

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java (hunk 1 above)
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java]: Beyond the scope of this branch -- if you think it should be optional, create a separate issue or PR.

File: src/test/java/org/broadinstitute/hellbender/utils/reference/ReferenceUtilsUnitTest.java (hunk 2 above)
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/utils/reference/ReferenceUtilsUnitTest.java]: Will add one

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java (hunk 1 above)
Comment (jean-philippe-martin) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java]: Fair enough. And that code's about to be changed anyways as it's integrated into the skeleton. I created issue #738 to track this.

**Comment by droazen** [comment]
Added bucket-based test case (and verified that it passes). @jean-philippe-martin please re-review

**Comment by jean-philippe-martin** [comment]
sorry, didn't see your note. However I'm sure the code's fine.

**Pull request merged**